src/Components/SingleArticle.jsx

```diff
@@ -1,28 +1,53 @@
 import { useState, useEffect } from 'react'
 import { useParams, Link, useNavigate } from 'react-router-dom'
-import { getArticleById } from "../api";
+import { getArticleById, getCommentsByArticleId } from "../api";
 import { Share, MessageCircle, ArrowLeft, ArrowBigUp, ArrowBigDown } from 'lucide-react';
 import { formatDate } from '../../utils/formatting';
+import CommentCard from './CommentCard';
 
 const SingleArticle = () => {
   const navigate = useNavigate();
   const { article_id } = useParams()
   const [article, setArticle] = useState([])
+  const [comments, setComments] = useState([])
+  const [isLoadingComments, setIsLoadingComments] = useState(true)
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState(null);
-
+  const [showComments, setShowComments] = useState(false);
+  
   useEffect(() => {
     setIsLoading(true);
-    getArticleById(article_id).then((articleData) => {
-      setArticle(articleData)
-      setIsLoading(false);
-    })
-    .catch((err) => {
-      console.error("Error fetching article:", err);
-      setError("Failed to load article");
-      setIsLoading(false);
-    });
-  }, [article_id])
+    getArticleById(article_id)
+      .then((articleData) => {
+        setArticle(articleData);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.error("Error fetching data:", err);
+        setError("Failed to load content");
+        setIsLoading(false);
+      });
+  }, [article_id]);
+
+  useEffect(() => {
```

Comment:
This could be extracted into it's own component `Comments` or something. That handles the fetching of the data then renders the comment cards